lib/watcher.ts

```diff
@@ -39,18 +38,36 @@ class Watcher extends EventEmitter {
     if (this.options.debounce == null) {
       this.options.debounce = 100;
     }
+    this._currentBuild = Promise.resolve();
     this.builder = builder;
     this.watcherAdapter =
       this.options.watcherAdapter || new WatcherAdapter(watchedNodes, this.options.saneOptions);
 
-    this.currentBuild = null;
     this._rebuildScheduled = false;
     this._ready = false;
     this._quittingPromise = null;
     this._lifetime = null;
     this._changedFiles = [];
   }
 
+  get currentBuild() {
+    return this._currentBuild;
+  }
+
+  private _updateCurrentBuild(promise: Promise<unknown>) {
+    this._currentBuild = promise;
+
+    promise.catch(() => {
+      /**
+       * The watcher internally follows currentBuild, and outputs errors appropriately.
+       * Since watcher.currentBuild is public API, we must allow public follows
+       * to still be informed of rejections.
+       *
+       * This function accomplishes that.
+       */
```

Comment:
I'm confused by this comment.  It looks to me that what it accomplishes is ensuring that a rejected `_currentBuild` is never an unhandled rejection.

The "this" that allowed public follows to handle rejections is that `_currentBuild === promise` rather than the chained promise from `catch`, but the comment reads to me that the "this" it's referring to is the catch handler itself.

```suggestion
      /**
       * The watcher internally follows currentBuild, and outputs errors appropriately.
       * Since watcher.currentBuild is public API, we must allow public follows
       * to still be informed of rejections.  However we do not want `_currentBuild` itself
       * to trigger unhandled rejections.
       *
       * By catching errors here, but returning `promise` instead of the chain from `promise.catch`,
       * both goals are accomplished.
       */
```